src/transaction/combine.ts

```diff
@@ -1,42 +1,63 @@
+
+/**
+ * The transactions should all be equal EXCEPT for the 'Signers' field. 
+ */
+ function validateAllSignedTransactionsAreEqual(transactions: Array<JsonObject>) {
+  const exampleTransaction = JSON.stringify({...transactions[0], Signers: null})
+  if (transactions.slice(1).some(tx => JSON.stringify({...tx, Signers: null}) !== exampleTransaction)) {
+    throw new ValidationError('txJSON is not the same for all signedTransactions')
+  }
+}
 
 function addressToBigNumber(address) {
   const hex = Buffer.from(decodeAccountID(address)).toString('hex')
   return new BigNumber(hex, 16)
 }
 
+/**
+ * According to the documentation on multi-signatures:
+ * If presented in binary form, the Signers array must be sorted based on 
+ * the numeric value of the signer addresses, with the lowest value first. 
+ * (If submitted as JSON, the submit_multisigned method handles this automatically.)
+ */
 function compareSigners(a, b) {
   return addressToBigNumber(a.Signer.Account).comparedTo(
     addressToBigNumber(b.Signer.Account)
   )
 }
 
+function getATransactionWithAllSigners(transactions: Array<JsonObject>): JsonObject {
+  // Signers must be sorted - see compareSigners for more details
+  const sortedSigners = _.flatMap(transactions, tx => tx.Signers)
+    .filter(signer => signer)
+    .sort(compareSigners)
+
+  return {...transactions[0], Signers: sortedSigners}
+}
+
+/**
+ * 
+ * @param signedTransactions A collection of the same transaction signed by different signers. The only difference
+ * between the elements of signedTransactions should be the Signers field.
+ * @returns An object with the combined transaction (now having a sorted list of all signers) which is encoded, along
+ * with a transaction id based on the combined transaction.
+ */
 function combine(signedTransactions: Array<string>): object {
   validate.combine({signedTransactions})
 
-  // TODO: signedTransactions is an array of strings in the documentation, but
-  // tests and this code handle it as an array of objects. Fix!
-  const txs: any[] = _.map(signedTransactions, binary.decode)
-  const tx = _.omit(txs[0], 'Signers')
-  if (!_.every(txs, (_tx) => _.isEqual(tx, _.omit(_tx, 'Signers')))) {
-    throw new utils.common.errors.ValidationError(
-      'txJSON is not the same for all signedTransactions'
-    )
-  }
-  const unsortedSigners = _.reduce(
-    txs,
-    (accumulator, _tx) => accumulator.concat(_tx.Signers || []),
-    []
-  )
-  const signers = unsortedSigners.sort(compareSigners)
-  const signedTx = _.assign({}, tx, {Signers: signers})
-  const signedTransaction = binary.encode(signedTx)
+  const transactions: JsonObject[] = signedTransactions.map(binary.decode);
+  validateAllSignedTransactionsAreEqual(transactions)
+
+  const oneTxWithAllSigners = getATransactionWithAllSigners(transactions)
+  const signedTransaction = binary.encode(oneTxWithAllSigners)
   const id = computeBinaryTransactionHash(signedTransaction)
   return {signedTransaction, id}
```

Comment:
i don't think you gain readability from assigning these variables. i think this is actually clearer, especially because the keys of the return are descriptive

```javascript
validateAllSignedTransactionsAreEqual(transactions)

const signedTransaction = binary.encode(getATransactionWithAllSigners(transactions))
return {
  id: computeBinaryTransactionHash(signedTransaction),
  signedTransaction,
}
```

Reply:
Yep, I'd agree with that. The function name makes it clear what it's doing.